File: lektion 1/app/src/main/java/com/example/helloworld/MainActivity.java
package com.example.helloworld;

import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);


            TextView tv = findViewById(R.id.label);
            tv.setText("Jag heter Aleksander");
            tv.setTextSize(26);

            Button b = findViewById(R.id.button);
            b.setText("tryck inte här");



            Button b2 = findViewById(R.id.button2);
            b2.setText("Hej");
            b2.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    b2.setText("Hejdå");
                    b2.setBackgroundColor(Color.RED);

                }
            });


            //skriv new och sen O så kommer det upp färdit
            b.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Log.i("Aleksander", "Click!!");

                }
            });

            b.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                b.setBackgroundColor(Color.BLACK);
                b.setText("Nu kommer du att dö");
                b.setTextSize(50);

                }
            });



            Log.println(Log.DEBUG, "Aleksander", "Hello world???");

            Toast.makeText(MainActivity.this, "Hello World", Toast.LENGTH_LONG).show();


            return insets;
        });
    }
}

File: Lektion 4/app/src/main/java/com/example/lektion4/MainActivity.kt
package com.example.lektion4

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    companion object{
        var defualtColor:String = "#ff00ff"
    }


    private lateinit var tv: TextView;
    public lateinit var x:Any

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        tv = findViewById(R.id.text)

        //variabler
        var foo = 5
        var bar:Int = 5 //datatypen
        var foo2:String = "hejsan"
        var decimalTal:Float = 0.5f //float
        var doubleTal:Double = 0.10000000001
        var longTal:Long = 1234455234234234L
        var c:Char = 'c'
        var bool:Boolean = false
        var x:Any = arrayOf("h","yo","Aleksander")

        //final
        val CONST_VALUE:Int = 2000;

        //int x=12,y=2,z=34 (java)
        var y:Int=5; var s:Int=6;

        var ia:IntArray = IntArray(5)
        var ia2:IntArray = intArrayOf(1,2,3,4,5)
        var ba:BooleanArray = BooleanArray(5)
        var ba2:BooleanArray = booleanArrayOf(true,false)
        var sa:Array<String> = arrayOf<String>("hej, tjena")

        if(1<5){
            Log.d("Aleksander", "hej ")
        } else
            Log.d("Aleksander", "hejdå: ")

        Log.d("Aleksander", if (1==1) "hej" else "hejdå")

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
}